src/oblivious/ristretto.py

```diff
@@ -385,6 +386,54 @@ def smu(s: bytes, t: bytes) -> bytes:
         """
         return _sc25519_mul(s, t)
 
+    @staticmethod
+    def sad(s: bytes, t: bytes) -> bytes:
+        """
+        Return the sum of two scalars.
+
+        >>> p = scalar.from_int(4)
+        >>> q = scalar.from_int(2)
+        >>> sodium.sad(p, q) == sodium.sad(q, p)
+        True
+        >>> sodium.sad(p, q).hex()
+        '0600000000000000000000000000000000000000000000000000000000000000'
+        >>> sodium.sad(-q, p).hex()
+        '0200000000000000000000000000000000000000000000000000000000000000'
+        """
+        (s, t) = (int.from_bytes(s, 'little'), int.from_bytes(t, 'little'))
+        return (
+                (s + t) % (pow(2, 252) + 27742317777372353535851937790883648493)
+        ).to_bytes(32, 'little')
+
+    @staticmethod
+    def ssu(s: bytes, t: bytes) -> bytes:
+        """
+        Return the result of subtracting the right-hand scalar from the
+        left-hand scalar.
+
+        >>> p = scalar.from_int(4)
+        >>> q = scalar.from_int(2)
+        >>> sodium.ssu(p, q).hex()
+        '0200000000000000000000000000000000000000000000000000000000000000'
+        """
+        (s, t) = (int.from_bytes(s, 'little'), int.from_bytes(t, 'little'))
+        return (
+                (s - t) % (pow(2, 252) + 27742317777372353535851937790883648493)
+        ).to_bytes(32, 'little')
+
+    @staticmethod
+    def sne(s: bytes) -> bytes:
+        """
+        Return the additive inverse of a scalar.
+
+        >>> p = scalar.from_int(4)
+        >>> sodium.sne(p).hex()
+        'e9d3f55c1a631258d69cf7a2def9de1400000000000000000000000000000010'
+        """
+        return (
+            (pow(2, 252) + 27742317777372353535851937790883648493 - int.from_bytes(s, 'little'))
```

Comment:
If we don't enforce the length of `s` to be in-range, I think there could be a bug unless we perform a modular reduction, via `%`, like the other two new scalar operation methods.